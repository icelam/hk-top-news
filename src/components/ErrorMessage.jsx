import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

/* Material UI Components */
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
    minHeight: '100vh'
  }
});

const ErrorMessage = ({ message }) => {
  const classes = useStyles();

  return (
    <Grid container direction="column" justify="center" alignItems="center" className={classes.root}>
      <Grid item>
        <Typography variant="body2" color="textSecondary" component="p" align="center">{message}</Typography>
      </Grid>
    </Grid>
  );
};

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired
};


export default ErrorMessage;
