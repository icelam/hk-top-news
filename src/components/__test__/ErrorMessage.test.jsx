import React from 'react';
import { render, screen } from '@testing-library/react';
import ErrorMessage from '../ErrorMessage';

const testMessage = 'Test Message';

describe('Error Message', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<ErrorMessage message={testMessage} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with correct error message', () => {
    render(<ErrorMessage message={testMessage} />);
    expect(screen.queryByText(testMessage)).toBeInTheDocument();
  });
});
