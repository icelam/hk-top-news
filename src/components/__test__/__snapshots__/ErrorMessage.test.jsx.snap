// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Message renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Test Message
      </p>
    </div>
  </div>
</DocumentFragment>
`;
